pkgs/by-name/sm/sm64coopdx/package.nix

```diff
@@ -0,0 +1,130 @@
+{
+  fetchFromGitHub,
+  lib,
+  writeTextFile,
+
+  bash,
+  curl,
+  hexdump,
+  python3,
+  SDL2,
+  stdenv,
+  zlib,
+
+  sm64baserom,
+  enableCoopNet ? true,
+  enableDiscord ? true,
+  enableTextureFix ? true,
+}:
+let
+  libc_hack = writeTextFile {
+    name = "libc-hack";
+    # https://stackoverflow.com/questions/21768542/libc-h-no-such-file-or-directory-when-compiling-nanomsg-pipeline-sample
+    text = ''
+      #include <unistd.h>
+      #include <string.h>
+      #include <pthread.h>
+    '';
+    destination = "/include/libc.h";
+  };
+  baserom = sm64baserom.override {
+    region = "us";
+    showRegionMessage = false;
+  };
+  pname = "sm64coopdx";
+  version = "1.0.3";
+in
+# note: there is a generic builder in pkgs/games/sm64ex/generic.nix that is meant to help build sm64ex and its forks; however sm64coopdx has departed significantly enough in its build that it doesn't make sense to use that other than the baseRom derivation
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "coop-deluxe";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-cIH3escLFMcHgtFxeSKIo5nZXvaknti+EVt72uB4XXc=";
+  };
+
+  buildInputs = [
+    curl
+    hexdump
+    libc_hack
+    python3
+    SDL2
+    zlib
+  ];
+
+  enableParallelBuilding = true;
+
+  makeFlags = [
+    "BREW_PREFIX=/not-exist"
+    "DISCORD_SDK=${if enableDiscord then "1" else "0"}"
+    "TEXTURE_FIX=${if enableTextureFix then "1" else "0"}"
+    "COOPNET=${if enableCoopNet then "1" else "0"}"
+  ];
+
+  preBuild = ''
+    # the baserom is needed both at build time and run time
+    ln -s ${baserom} baserom.us.z64
+    # remove -march flags, stdenv manages them
+    substituteInPlace Makefile \
+      --replace-fail ' -march=$(TARGET_ARCH)' ""
+    # workaround a bug in the build
+    # see https://github.com/coop-deluxe/sm64coopdx/issues/186#issuecomment-2216163935
+    # this can likely be removed when the next version releases
+    make build/us_pc/sound/sequences.bin
+  '';
+
+  installPhase = ''
+    runHook preInstall
```

Comment:
should be fine to change `installPhase = ...` to `postInstall = ...` and remove the `runHook` calls.

Reply:
I don't think that's right; I want to replace the installPhase, not run something after it.